scripts/build-and-test.sh

```diff
@@ -19,10 +19,21 @@ function run_tests_with_gdb {
   for x in bin/**/*-tests; do scripts/run-with-gdb.sh "$x"; done
 }
 
+function run_tests {
+    for x in bin/**/*-tests; do "$x"; done
+}
+
+num_procs=1
+if [[ "$(uname)" == "Darwin" ]]; then
+    num_procs=$(sysctl -n hw.ncpu)
+elif [[ "$(expr substr $(uname -s) 1 5)" == "Linux" ]]; then
+    num_procs=$(lscpu -p | grep -v '^#' | sort -u -t, -k 2,4 | wc -l) # pysical cores
```

Comment:
`nproc` works in Docker / linux

Above method shows 32 on CircleCI, when really the container only has  2 `available`